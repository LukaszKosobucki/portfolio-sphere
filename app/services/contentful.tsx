import { createClient } from "contentful";

export const fetchSpecificEntries = async ({
  content_type,
  title,
  multiple = false,
  order,
}: {
  content_type: string;
  title?: string;
  multiple?: boolean;
  order?: string;
}) => {
  const client = createClient({
    space: process.env.CONTENTFUL_SPACE_ID ?? "",
    environment: "master", // defaults to 'master' if not set
    accessToken: process.env.CONTENTFUL_ACCESS_TOKEN ?? "",
  });

  const query = {
    content_type,
    ...(title && { "fields.title": title }),
    limit: multiple ? 100 : 1,
    order,
  };

  const entries = await client.getEntries(query).then((res) => res.items);

  return entries;
};
