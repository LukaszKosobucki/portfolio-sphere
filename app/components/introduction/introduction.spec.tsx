import { fetchSpecificEntries } from "@/app/services/contentful";
import { formatSnapshot } from "@/app/utils/tests/format-snapshot";
import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
import { render } from "@testing-library/react";
import { afterAll, describe, expect, jest, mock, test } from "bun:test";
import Introduction from "./introduction";

mock.module("@/app/services/contentful", () => ({
  fetchSpecificEntries: jest.fn(),
}));

mock.module("@contentful/rich-text-react-renderer", () => ({
  documentToReactComponents: jest.fn(),
}));

describe("Introduction Component", () => {
  const mockResponse = [
    {
      fields: {
        description: { content: "Mock description content" },
      },
    },
  ];

  afterAll(() => {
    jest.clearAllMocks();
  });

  test("Should match snapshot", async () => {
    (fetchSpecificEntries as jest.Mock).mockResolvedValue(mockResponse);
    (documentToReactComponents as jest.Mock).mockReturnValue(<p>Mocked rich text content</p>);
    const { asFragment } = render(await Introduction());

    expect(formatSnapshot(asFragment())).toMatchSnapshot();
  });

  test("fetchSpecificEntries is called with correct arguments", async () => {
    (fetchSpecificEntries as jest.Mock).mockResolvedValue(mockResponse);
    render(await Introduction());

    expect(fetchSpecificEntries).toHaveBeenCalledWith({
      content_type: "description",
      title: "About me",
    });
  });
});
