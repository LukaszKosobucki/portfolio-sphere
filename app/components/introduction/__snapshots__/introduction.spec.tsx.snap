// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction Component Should match snapshot 1`] = `
"<DocumentFragment>
  <div
    class="relative bg-white dark:bg-black"
  >
    <div
      class="mx-auto max-w-screen-sm p-5"
    >
      <section
        class="animate"
      >
        <article
          class="text-justify"
        >
          <p>
            Mocked rich text content
          </p>
        </article>
      </section>
    </div>
  </div>
</DocumentFragment>"
`;
